File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min=Integer.MAX_VALUE;
        int total=0;
        for(int price:prices){
            if (price<min){
                min=price;
            }else{
                total=Math.max(total,price-min);

            }
            }
            

        
        return total;
        

    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice=float(inf)
        maxprofit=0

        for price in prices:
            if price<minprice:
                minprice=price

            profit=price-minprice

            if profit>maxprofit:
                maxprofit=profit

        return maxprofit  